### PR TITLE
DP-857 handle 1-n input topics

### PR DESCRIPTION
**Pull request by viktort**
👷 ⚠️ 👷 
Addresses issue #12 

Some feedback would be great 👍

### CODE REVIEW

**Review by michaelsmithhx — COMMENTED** [review]
What's the intended behaviour here? As it stands, if the input topic runs through cleanly once and then fails on a subsequent iteration, the pipeline can still be setup but in an incomplete state. Ideally the method should be split as its getting a bit long.

**Comment by viktort** [comment]
@michaelsmithhx I agree, there is scope to break up the `main` method. The aim here is to ensure that we can pass in list of input topics for the pipeline to read from. From @LusciousPear comment, it is pointing towards the configuration not being correct in the this case.

**Comment by viktort** [comment]
Going to close this for now until I have investigated further 👍

**Comment by viktort** [comment]
Needs further testing but I have ran locally and I get the pipeline connecting to multiple topics.

File: src/main/java/com/acacia/angleddream/Main.java [hunk 1]
```diff
@@ -132,56 +137,58 @@ public static void main(String[] args) throws IOException {
 
         if (!outputTopics.isEmpty()) {
 
-            try {
+            for(String inputTopic: inputTopics) {
+                try {
 
-                PCollection<String> inp = pipeline.apply(PubsubIO.Read.topic(options.getPubsubTopic()));
+                    PCollection<String> inp = pipeline.apply(PubsubIO.Read.topic(inputTopic));
```
Comment (ekhaydarov) [inline src/main/java/com/acacia/angleddream/Main.java:144]: `PCollection<String> inp = pipeline.apply(PubsubIO.Read.topic(inputTopic));` move this outside the try and remove `for(String inputTopic: inputTopics)`.

Then create a ParDo to launch as many parallel topic reads as required which all end up in one `PCollection<String> inp`

The rest of the code will then not need to change.

**Comment by LusciousPear** [comment]
OK, looks good.

**Pull request merged**